code/modules/station_goals/shield.dm

```diff
@@ -168,9 +169,10 @@
 			change_meteor_chance(0.5)
 
 /obj/machinery/satellite/meteor_shield/proc/change_meteor_chance(mod)
-	var/datum/round_event_control/E = locate(/datum/round_event_control/meteor_wave) in SSevents.control
-	if(E)
-		E.weight *= mod
+	// Update the weight of all meteor events
+	for(var/datum/round_event_control/meteor_wave/meteors in SSevents.control)
+		if(istype(meteors))
+			meteors.weight *= mod
```

Comment:
```suggestion
	for(var/datum/round_event_control/meteor_wave/meteors in SSevents.control)
		meteors.weight *= mod
```
You are already type checking in the `for` loop. This check can never be false.

Reply:
removed that extra check now.